tests/subset_write_test.py

```diff
@@ -0,0 +1,110 @@
+from tests.helper import requires
+
+import numpy as np
+from swiftsimio.subset_writer import write_subset, find_datasets
+import swiftsimio as sw
+import h5py
+import sys
+
+def compare_arrays(A, B):
+    """
+    Compares two arrays or arrays of arrays
+
+    Parameters
+    ----------
+    A, B : np.array
+        Arrays to compare
+
+    Returns
+    -------
+    bool 
+        True if all elements are identical between `A` and `B`
+    """
+    # Check we're not going crazy
+    if len(A) != len(B):
+        return False
+    
+    # Compare element-by-element
+    for i in range(len(A)):
+        try:
+            if A[i] != B[i]:
+                return False
+        except:
+            if not all(A[i][:] == B[i][:]):
+                return False
+
+    # All good
+    return True
```

Comment:
I'm confused here why `(A == B).all()` is not used - am I missing something?

Reply:
I tried this, didn't work. I think this is because of the datasets which contain an array of arrays.

Reply:
That should work - the maintainers of numpy are a lot smarter than us :).

Can you provide an example that does not work?